### DISCUSSION TITLE
I don't know what GitHub is about

### DISCUSSION THREAD

**Discussion by shedhand** [community/community#138085] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hello, I hope I don't get banned, as I haven't visited GitHub for many years and no longer know how it works or understand the jargon.
I downloaded the tar.gz file for DIY Layout Creator on my Linux Ubuntu desktop but I have no idea how to install it, nor can I find any installation info on the World Wide Internet thingy. I'm an old man losing my marbles.
Can anyone point me in the right direction for step by step, how to install the program please...please?

**Comment by Zouhair180** [comment]
Chahad hhh

**Comment by ferid333** [comment]
1. **Open Terminal:**  
   First, open your terminal. You can do this by pressing `Ctrl + Alt + T`.

2. **Navigate to the Download Directory:**  
   Use the terminal to go to the folder where the tar.gz file is saved. Usually, it’s in your Downloads folder. You can type:
   ```
   cd ~/Downloads
   ```

3. **Extract the tar.gz File:**  
   Now, extract the contents of the file. Use this command:
   ```
   tar -xvzf filename.tar.gz
   ```
   Replace `filename.tar.gz` with the actual name of your file.

4. **Enter the Extracted Folder:**  
   After extraction, navigate into the new folder created from the tar.gz file:
   ```
   cd foldername
   ```
   Replace `foldername` with the name of the folder that was extracted.

5. **Look for an Installation Guide:**  
   Inside the folder, look for a `README` or `INSTALL` file. You can open these files using a text editor like `nano` or just view them in the terminal:
   ```
   nano README
   ```

6. **Run the Program (if no installation is needed):**  
   If there’s no complicated installation, you might be able to run the program directly. Look for an executable file and type:
   ```
   ./filename
   ```
   Replace `filename` with the name of the executable.

7. **Install Dependencies (if needed):**  
   If the program requires extra software, the instructions should tell you what to install. You can install packages using:
   ```
   sudo apt install packagename
   ```

If you get stuck at any step, feel free to ask for help again! You're doing great.

**Comment by shedhand** [comment]
Many thanks my friend.😎

**Reply by ferid333** [reply]
You are welcome please just close the discussion by choosing my reply as answer

**Reply by shedhand** [reply]
Sry. But my head is done.
I tried what you said but got stuck at your Step 4. I tried multiple times.
Thanks

mike@mike-ET2321I:~$ cd ~/Downloads
mike@mike-ET2321I:~/Downloads$ tar -diylayoutcreator.tar.gz
tar: invalid option -- 'y'
Try 'tar --help' or 'tar --usage' for more information.
mike@mike-ET2321I:~/Downloads$ diylayoutcreator.tar.gz
diylayoutcreator.tar.gz: command not found
mike@mike-ET2321I:~/Downloads$ tar -xvzf diylayoutcreator.tar.gz
tar (child): diylayoutcreator.tar.gz: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
mike@mike-ET2321I:~/Downloads$

**Comment by nandamade** [comment]
nice question
